lib/server/routes/buckets.js

```diff
@@ -1396,6 +1399,109 @@ BucketsRouter.prototype.getStorageUsage = function (req, res) {
   });
 };
 
+BucketsRouter.prototype.startUpload = function (req, res, next) {
+  const { Bucket, Upload, Contact } = this.storage.models;
+
+  const { uploads } = req.body;
+
+  if (!uploads) {
+    return next(new errors.BadRequestError('Missing uploads'));
+  }
+
+  if (!Array.isArray(uploads)) {
+    return next(new errors.BadRequestError('Uploads is not an array'));
+  }
+
+  Bucket.findOne({
+    _id: req.params.id,
+    user: req.user._id
+  }, (err, bucket) => {
+    if (err) {
+      return next(new errors.InternalError(err.message));
+    }
+
+    if (!bucket) {
+      return next(new errors.NotFoundError('Bucket not found'));
+    }
+
+    const response = [];
+
+    async.eachLimit(
+      uploads,
+      1,
+      (upload, cb) => {
+        const { index, size } = upload;
+
+        if (typeof size !== 'number' || size < 0) {
```

Comment:
Move all validations to the beginning (using a for or something like that) in order to:
- Avoid inserting an upload before ensuring all the entire data is ok. If the 3rd upload is not okay, you will have two uploads stalled
- Having the validations located in one point, thus, at the time of starting to use layers, will be easier to decouple validations from business logic and / or persistence

Reply:
Agree, on it 👍

Reply:
fixed